src/util/CancellationHandle.h

```diff
@@ -33,16 +33,18 @@ enum class CancellationState {
 
 namespace detail {
 
-#if !defined(QLEVER_DISABLE_QUERY_CANCELLATION_WATCH_DO) && LOGLEVEL >= WARN
-constexpr bool ENABLE_QUERY_CANCELLATION_WATCH_DOG = true;
+/// Helper enum that selects which features to compile inside the
+/// `CancellationHandle` class.
+enum class CancellationMode { ENABLED, NO_WATCH_DOG, DISABLED };
+
+constexpr CancellationMode CANCELLATION_MODE = []() {
```

Comment:
Add a comment about turning the macro that comes from cmake itno a proper `contexpr` variable.